cpp/CMakeLists.txt

```diff
@@ -215,6 +215,27 @@ else()
     message(FATAL_ERROR "Boost not found, please check your settings.")
 endif(Boost_FOUND)
 
+###################################################################################################
+# - cuFile -----------------------------------------------------------------------------------
+
+find_path(CUFILE_INCLUDE "cufile.h"
+    HINTS "${GDS_ROOT}/lib"
+          "/usr/local/cuda/lib64")
+
+find_library(CUFILE_LIBRARY "libcufile.so"
+    HINTS "${GDS_ROOT}/lib"
+          "/usr/local/cuda/lib64")
+
+message(STATUS "CUFILE: CUFILE_LIBRARY set to ${CUFILE_LIBRARY}")
+message(STATUS "CUFILE: CUFILE_INCLUDE set to ${CUFILE_INCLUDE}")
```

Comment:
We may want to move this to a `FindcuFile` cmake module or something similar and create a target instead of using `CUFILE_LIBRARY` / `CUFILE_INCLUDE` variables.

Reply:
But lets get everything else squared away first and we can tackle this before merging.